src/Steps/Release/MergeUpRelease.php

```diff
@@ -0,0 +1,181 @@
+        $currentBranchIsRelease = preg_match(
+            '/' . preg_quote(RewriteReleaseBranches::RELEASE_SUFFIX) . '$/',
+            $currentBranch
+        );
+        if (!$currentBranchIsRelease) {
+            $this->log(
+                $output,
+                "Warning: Current branch '$currentBranch' is not a release branch. "
+                    . "Skipping $currentLibrary and its children.",
+                'bg=yellow'
+            );
+            return;
+        }
+
+        // Do the merge up recursively
+        foreach ($releasePlanNode->getItems() as $item) {
+            $this->mergeRecursive($output, $item, $deleteBranch);
+        }
+        $this->mergeLibrary($output, $releasePlanNode);
+
+        // Optionally delete the branch from the remote
+        if ($deleteBranch) {
+            $this->log(
+                $output,
+                "Deleting branch <info>$currentBranch</info> from remote for <info>{$currentLibrary}</info>"
+            );
+            $releasePlanNode->getLibrary()->getRepository()->run('push', ['--delete', $currentBranch]);
+        }
+    }
+
+    /**
+     * Performs a merge up for a single library
+     *
+     * @param OutputInterface $output
+     * @param LibraryRelease $releasePlanNode Node in release plan being released
+     */
+    protected function mergeLibrary(OutputInterface $output, LibraryRelease $releasePlanNode)
+    {
+        $library = $releasePlanNode->getLibrary();
+        $releaseBranch = $library->getBranch();
+        $minorBranch = str_replace(RewriteReleaseBranches::RELEASE_SUFFIX, '', $releaseBranch);
+        $majorBranch = $releasePlanNode->getVersion()->getMajor();
+
+        // Merge up to minor branch, then to major branch
+        $this->mergeUp($output, $releasePlanNode, $releaseBranch, $minorBranch, false);
+        $this->mergeUp($output, $releasePlanNode, $minorBranch, $majorBranch, true);
+
+        // Always check out the release branch at the end
+        $library->checkout($output, $releaseBranch);
+    }
+
+    /**
+     * Merge up from one branch to another, setting the appropriate constraints for released dependencies as we go.
+     */
+    protected function mergeUp(
+        OutputInterface $output,
+        LibraryRelease $releasePlanNode,
+        string $mergeFrom,
+        string $mergeInto,
+        string $isMajorBranch
+    ) {
+        $library = $releasePlanNode->getLibrary();
+        $name = $library->getName();
+
+        // Step 1. Checkout branch to merge into
+        $library->checkout($output, $mergeInto);
+
+        $this->log(
+            $output,
+            "Merging library <info>{$name}</info> from <info>{$mergeFrom}</info> to <info>{$mergeInto}</info>"
+        );
+
+        // Step 2. Merge up if there's anything to merge
+        if ($library->hasDiff($output, $mergeFrom)) {
+            // Note: If there are conflicts or other issues merging,
+            // a Gitonomy\Git\Exception\ProcessException will be thrown
```

Comment:
```suggestion
            // a Gitonomy\Git\Exception\ProcessException will be thrown and cow will exit
```
So, lets say this happens ... what now?

Reply:
The release manager should resolve the conflict, then checkout the release branch again, and re-run the command.

Cow will still go through and process the branches it's already done - but it will find there is nothing to merge, and the composer changes will match what's already there, so no new commits will be added, so it won't cause any changes to be made on the modules which were already processed.
It will then get to the one where the release manager resolved the conflicts and continue from there.